frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -343,7 +344,20 @@ impl<T: Config> Pallet<T> {
 	///
 	/// This drains the [`SignedSubmissions`], potentially storing the best valid one in
 	/// [`QueuedSolution`].
-	pub fn finalize_signed_phase() -> (bool, Weight) {
+	///
+	/// This is a *self-weighing* function, it automatically registers its weight internally when
+	/// being called.
+	pub fn finalize_signed_phase() -> bool {
+		let (weight, found_solution) = Self::finalize_signed_phase_internal();
+		<frame_system::Pallet<T>>::register_extra_weight_unchecked(
```

Comment:
I think we re-use `register_weight` here?

Reply:
This can only be allowed when being called from `on_initialize`

Otherwise, all other kinds of weight registration need to happen before things start to execute.

Reply:
the entire approach of this PR is assuming mandatory weight @shawntabrizi 

`register_weight` can be reused here as well @emostov 👍